File: lib/src/wisepos/wisepos_device.dart
import 'package:wiseasy_sdk/src/generated/wiseasy_sdk.g.dart';

class WisePosDevice {
  WisePosDevice._();
  static WisePosDevice? _instance;
  static WisePosDevice get instance => _instance ??= WisePosDevice._();

  final WisePosDeviceChannel _channel = WisePosDeviceChannel();

  Future<Map<String, String>> getVersionInfo() async =>
      Map<String, String>.from(await _channel.getVersionInfo());

  Future<Map<String, String>> getKernelVersion() async =>
      Map<String, String>.from(await _channel.getKernelVersion());

  Future<Map<String, int>> getTamperStatus() async =>
      Map<String, int>.from(await _channel.getTamperStatus());

  Future<String> getDeviceSn() => _channel.getDeviceSn();
}

File: android/src/main/kotlin/com/visionpay/wiseasy_sdk/wisepos/WisePosDevice.kt
package com.visionpay.wiseasy_sdk.wisepos

import com.visionpay.wiseasy_sdk.WisePosDeviceChannel
import com.wisepos.smartpos.device.Device

class WisePosDevice(private val device: Device) : WisePosDeviceChannel {

    override fun getVersionInfo(): Map<String, String> {
        return device.versionInfo
    }

    override fun getKernelVersion(): Map<String, String> {
        return device.kernelVersion
    }

    override fun getTamperStatus(): Map<String, Long> {
        return device.tamperStatus.mapValues {
            return mapOf(it.key to it.key.toLong())
        }
    }

    override fun getDeviceSn(): String {
        return device.deviceSn
    }
}